react.js

```diff
@@ -1,30 +1,43 @@
 import {writable, get} from 'svelte/store';
-import {beforeUpdate} from 'svelte';
+import {afterUpdate} from 'svelte';
 
 let currentStore;
 let i = 0;
+let effects = [];
+let layoutEffects = [];
 
 export function wrap(fn) {
   let store = writable([]);
-  currentStore = store;
-  i = 0;
-
-  let res = fn();
-
   let stores = {};
-  for (let key in res) {
-    stores[key] = writable(res[key]);
-  }
+  
+  let _effects = effects;
+  let _layoutEffects = layoutEffects;
+
+  afterUpdate(() => {
+    _effects.forEach(fn => fn());
+  });
 
   store.subscribe(function () {
     currentStore = store;
     i = 0;
+    effects = [];
+    layoutEffects = [];
 
     let res = fn();
     for (let key in res) {
-      stores[key].set(res[key]);
+      if (!(key in stores)) {
+        stores[key] = writable(res[key]);
+      } else {
+        stores[key].set(res[key]);
+      }
     }
 
+    _effects = effects;
+    _layoutEffects = layoutEffects;
+    requestAnimationFrame(() => {
+      _layoutEffects.forEach(fn => fn());
```

Comment:
I think this one should be effects? layout effects are sync, effects are async.

```suggestion
      _effects.forEach(fn => fn());
```

Maybe put the `requestAnimationFrame` inside the `afterUpdate` block as well so all the effects are together?

Reply:
oops. my bad. fixed it.